src/EditorFeatures/Core/Implementation/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -122,6 +122,12 @@ private string GetRootPathWithGuid_NoLock()
                         }
                         catch (Exception e) when (FatalError.ReportWithoutCrashUnlessCanceled(e))
                         {
+                            var failureText = e.ToString();
+                            var failureTextAsComment = "// " + failureText.Replace("\n", "\n// ") + "\n";
+
+                            var sourceText = await temporaryDocument.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                            temporaryDocument = temporaryDocument.WithText(sourceText.Replace(default, failureTextAsComment));
```

Comment:
The "default" there totally threw me for a loop. Can we just say new TextSpan(0, 0) to make it clear what the intent is? My brain's first binding there was this was a call to string.Replace and thus the default was a variable somewhere else...

Also, contractually, sourceFromMetadataService.AddSoruceToAsync() is more or less allowed to overwrite the contents of whatever you put there (the assumption being that going in it's empty.) If the code is doing "insert the generated stuff at location zero", that's just happenstance that it works.

Reply:
(for the latter point: just adding a comment somewhere to make your intent clear that you're relying on it is more than fine by me)

Reply:
Added a comment, changed to `new TextSpan(0, 0)`, and added a test to verify the behavior.